controller/array_action/array_mediator_svc.py

```diff
@@ -517,14 +517,23 @@ def _stop_and_delete_fcmap(self, fcmap_id):
         self._delete_fcmap(fcmap_id, force=True)
 
     def _safe_delete_fcmaps(self, object_name, fcmaps):
-        unfinished_fcmaps = [fcmap for fcmap in fcmaps
-                             if fcmap.status != FCMAP_STATUS_DONE or fcmap.copy_rate == "0"]
-        if unfinished_fcmaps:
-            raise array_errors.ObjectIsStillInUseError(id_or_name=object_name,
-                                                       used_by=unfinished_fcmaps)
+        fcmaps_to_delete = []
+        fcmaps_in_use = []
+
         for fcmap in fcmaps:
+            if not self._is_in_remote_copy_relationship(fcmap):
+                if fcmap.status != FCMAP_STATUS_DONE or fcmap.copy_rate == "0":
```

Comment:
should we wrap the `fcmaps_in_use` checks with a function as well?

Reply:
I don't mind wrapping this code, but I think we should have something like:
```suggestion
                if self._is_fcmap_in_use(fcmap):
                    ...
            def _is_fcmap_in_use(self, fcmap):
                 return self._is_fcmap_active(fcmap) or self._is_fcmap_relate_to_snapshot(fcmap)
```
WDYT?

Reply:
I'm not sure `relate_to_snapshot` is accurate enough, since volume from snapshot can be active, but would get False if theoretically would be passed to `_is_fcmap_relate_to_snapshot`

Reply:
and I assume you meant `related`

Reply:
yes, it's a typo.
What's your suggestion for wrapping the code?

Reply:
maybe `_is_fcmap_relate_to_snapshot` -> `_is_snapshot_fcmap`?

Reply:
I'm not sure we do things like this (non-related code improvements) in PRs.
Do you want to open a ticket for that?

Reply:
no, I suggested this because you wrote that you [don't mind wrapping this code](https://github.com/IBM/ibm-block-csi-driver/pull/352#discussion_r685223423). if you think it shouldn't be part of this PR, then I will resolve

Reply:
I wrote that because I thought we want to improve the places we touch anyway.
Then I realized [we don't](https://github.com/IBM/ibm-block-csi-driver/pull/355#discussion_r685778214).

Reply:
there is no "we" in this link - if both sides agree to include a change, it's allowed.
some suggestions are just not merge blockers, that's all

Reply:
@ArbelNathan likes to keep the PR without unrelated code improvements. That's the reason I feel it's a team decision instead of an individual one.

Reply:
then in my sentence replace `both sides` with `all sides`

Reply:
@ArbelNathan would it be ok to include the suggested wrap in this PR? if not, I will resolve (even if most of us are ok with including it)

Reply:
I believe that any improvement that related to the places that the diff is touching is worth doing. but as @zingero [linked](https://github.com/IBM/ibm-block-csi-driver/pull/352#discussion_r689361028), it seems that the implementation of that notion is very selective..

Reply:
great, so it seems in this case we are all ok with the wrapping suggestion

Reply:
Actually, that's not the case.

Reply:
I'm not asking if it's relevant, I'm asking if **in this case** we are ok with including an "irrelevant" change.
not all cases are the same

Reply:
I'm not ok with it

Reply:
I got a [different impression](https://github.com/IBM/ibm-block-csi-driver/pull/352#discussion_r689352298), but ok